dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/TeradataQueryLogsJdbcTask.java

```diff
@@ -0,0 +1,149 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.connector.teradata;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import com.google.common.io.ByteSink;
+import com.google.edwmigration.dumper.application.dumper.handle.JdbcHandle;
+import com.google.edwmigration.dumper.application.dumper.task.AbstractJdbcTask;
+import com.google.edwmigration.dumper.application.dumper.task.TaskRunContext;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.sql.Clob;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Base64;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.jdbc.core.ResultSetExtractor;
+
+public class TeradataQueryLogsJdbcTask extends AbstractJdbcTask<Void> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TeradataQueryLogsJdbcTask.class);
+  private final DateTimeFormatter SQL_FORMAT =
+      DateTimeFormatter.ISO_OFFSET_DATE_TIME.withZone(ZoneOffset.UTC);
+  private ZonedDateTime queryLogStartDate, queryLogEndDate;
+  private final String tableName;
+
+  public TeradataQueryLogsJdbcTask(
+      String targetPath,
+      String tableName,
+      ZonedDateTime queryLogStartDate,
+      ZonedDateTime queryLogEndDate) {
+    super(targetPath);
+    this.tableName = tableName;
+    this.queryLogStartDate = queryLogStartDate;
+    this.queryLogEndDate = queryLogEndDate;
+  }
+
+  @CheckForNull
+  @Override
+  protected Void doInConnection(
+      @Nonnull TaskRunContext context,
+      @Nonnull JdbcHandle jdbcHandle,
+      @Nonnull ByteSink sink,
+      @Nonnull Connection connection)
+      throws SQLException {
+    LOG.info("Getting first and last query log entries");
+
+    String[] queryLogsFirstAndLastEntries = new String[2];
+    String sql = sqlForQueryLogDates(tableName, queryLogStartDate, queryLogEndDate);
+    doSelect(connection, newCsvResultSetExtractor(sink, queryLogsFirstAndLastEntries), sql);
+
+    LOG.info(
+        "The first query log entry is {} UTC and the last query log entry is {} UTC",
+        queryLogsFirstAndLastEntries[0],
+        queryLogsFirstAndLastEntries[1]);
+
+    return null;
+  }
+
+  private ResultSetExtractor<Void> newCsvResultSetExtractor(
+      @Nonnull ByteSink sink, String[] queryLogsFirstAndLastEntries) {
```

Comment:
I'd prefer it if this extractor had a return value rather than return `Void` and modify its parameter.